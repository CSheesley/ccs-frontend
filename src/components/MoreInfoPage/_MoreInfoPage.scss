.faq-container {
  background: $muted-blue;
  color: $navy;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 20px;
  margin-top: 4rem;
}

.faq-question {
  font-size: 24px;
  cursor: pointer;
  color: $navy;
  width: 90%;
  text-transform: lowercase;
  &:hover {
    color: $white;
  }
  &:first-letter {
    text-transform: uppercase;
  }
}

.faq-question::before {content: "â€¢"; color: hsl(0, 0%, 0%)};

.answer-modal {
  color: hsl(203, 52%, 19%);
  width: 80%;
  border-radius: 6px;
  z-index: 5;
  padding: 20px;
  position: absolute;
  background: white;
  line-height: 1.4;
}



.x-button {
  height: 20px;
  width: 20px;
  cursor: pointer;
  position: absolute;
  top: 0;
  right: 0;
  z-index: 5;
  margin: 10px;
}