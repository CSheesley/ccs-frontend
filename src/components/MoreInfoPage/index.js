import React, { useState } from 'react';
import { faqs } from '../../assets/staticData/staticFAQ';

const MoreInfoPage = () => {
  const [answer, setAnswer] = useState('');
  const [pageY, setPageY] = useState(0);

  const questions = faqs.map((faq) => {
    return <p
      className="faq-question"
      role="button"
      aria-pressed="false"
      key={faq.id}
      onClick={
        (e) => {
          e.persist();
          setAnswer(faq.answer);
          setPageY(e.pageY);
        }
      }
    >
      {faq.question}
    </p>
  });


  return (
    <section className="faq-container">
      {questions}
      {
        answer
        &&
        <div className="answer-modal" style={{ transform: `translate(0, ${pageY}px)` }} role="dialog" aria-labelledby="dialog1Title" aria-describedby="dialog1Desc">
          <h2 id="dialog1Title">Answer:</h2>
          <p id="dialog1Desc" className="faq-answer">{answer}</p>
          <img
            className="x-button"
            alt="close answer modal icon"
            role="button" 
            src={require("../../assets/images/icons/cancel-x.svg")}
            onClick={ () => setAnswer('') }
          />
        </div>
      }
    </section>
  );
};

export default MoreInfoPage;