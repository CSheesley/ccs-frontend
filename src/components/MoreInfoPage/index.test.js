import React from 'react';
import { shallow } from 'enzyme';
import MoreInfoPage from '.';


describe('MoreInfoPage', () => {
  let wrapper;

  it('should match the snapshot', () => {
    wrapper = shallow(<MoreInfoPage />);

    expect(wrapper).toMatchSnapshot();
  });
});