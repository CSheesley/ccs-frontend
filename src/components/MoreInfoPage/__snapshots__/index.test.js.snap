// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoreInfoPage should match the snapshot 1`] = `ShallowWrapper {}`;
